src/main/java/appeng/util/item/MeaningfulItemIterator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.item;
+
+
+import java.util.Iterator;
+import java.util.NavigableMap;
+
+import appeng.api.storage.data.IAEItemStack;
+
+
+public class MeaningfulItemIterator<T extends IAEItemStack> implements Iterator<T>
```

Comment:
`FluidList` has its own iterator as they are no longer share the same implementation. Thus `IAEItemStack` is perfectly valid.